VideoLocker/src/main/java/org/edx/mobile/event/NewVersionAvailableEvent.java

```diff
@@ -0,0 +1,193 @@
+        this.isUnsupported = isUnsupported;
+    }
+
+    /**
+     * @return The version number of the latest release of the app, or {@code null} if not
+     *         available.
+     */
+    @Nullable
+    public ArtifactVersion getNewVersion() {
+        return newVersion;
+    }
+
+    /**
+     * @return The last date on which the current version of the app will be supported, or
+     *         {@code null} if not available.
+     */
+    @Nullable
+    public Date getLastSupportedDate() {
+        return lastSupportedDate;
+    }
+
+    /**
+     * Returns whether the current version is unsupported. This is based on whether we're getting
+     * HTTP 426 errors, and thus can't be inferred from the last supported date (the two properties
+     * may not be consistent with each other due to wrong local clock time or an inconsistency in
+     * the server configurations).
+     *
+     * @return Whether the current version is unsupported.
+     */
+    public boolean isUnsupported() {
+        return isUnsupported;
+    }
+
+    /**
+     * Resolve the notification string, and return it.
+     *
+     * @param context A Context to resolve the string
+     * @return The notification string.
+     */
+    @NonNull
+    public CharSequence getNotificationString(@NonNull final Context context) {
+        @StringRes
+        final int notificationStringRes;
+        if (isUnsupported) {
+            notificationStringRes = R.string.app_version_unsupported;
+        } else if (lastSupportedDate == null) {
+            notificationStringRes = R.string.app_version_outdated;
+        } else {
+            // Deadline date is available, but won't be displayed for now.
+            notificationStringRes = R.string.app_version_deprecated;
+        }
+        return context.getText(notificationStringRes);
+    }
+
+    /**
+     * Mark the event as consumed by the subscribers.
+     */
+    public void markAsConsumed() {
+        isConsumed = true;
+    }
+
+    /**
+     * @return Whether the event has been consumed by the subscribers.
+     */
+    public boolean isConsumed() {
+        return isConsumed;
+    }
+
+    /**
+     * Compare this to another instance to determine their priority. Events reporting the current
+     * app version as unsupported have the highest priority, followed by deprecation events, which
+     * are prioritised according to the closeness of the last supported date they report, followed
+     * by new version availability events, which are prioritized according to the reported new
+     * version number.
+     *
+     * @param another the object to compare to this instance.
+     * @return a negative integer if this instance has lesser priority than {@code another};
+     *         a positive integer if this instance has greater priority than {@code another};
+     *         0 if this instance has the same priority as {@code another}.
+     * @throws ClassCastException if {@code another} cannot be converted into
```

Comment:
Good point. I used the javadoc from the `Comparable<T>` as a template which doesn't have it's generic type defined, so it has this clause. As long as you're calling this specific method as defined without unsafely casting it to a non-generic `Comparable` type, it should only ever receive the instances of the same class as the parameter in the `compareTo()` method, and thus no `ClassCastException` should ever be thrown. I have removed this javadoc declaration in commit ee0d62e006addbb4334b5490edc5da0cf3a58282.